src/basic/ether-addr-util.h

```diff
@@ -1,11 +1,31 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
 #pragma once
 
+#include <linux/if_infiniband.h>
 #include <net/ethernet.h>
 #include <stdbool.h>
 
 #include "hash-funcs.h"
 
+#define HW_ADDR_MAX_SIZE CONST_MAX(ETH_ALEN, INFINIBAND_ALEN)
+
+union hw_addr_union {
+        struct ether_addr ether;
+        uint8_t infiniband[INFINIBAND_ALEN];
+        uint8_t bytes[HW_ADDR_MAX_SIZE];
+};
+
+typedef struct hw_addr_data {
+        union hw_addr_union addr;
+        size_t length;
+} hw_addr_data;
+
+#define HW_ADDR_TO_STRING_MAX (3*HW_ADDR_MAX_SIZE)
+char* hw_addr_to_string(const hw_addr_data *addr, char buffer[HW_ADDR_TO_STRING_MAX]);
+
+/* Use only as function argument, never stand-alone! */
+#define HW_ADDR_TO_STR(hw_addr) hw_addr_to_string((hw_addr), (char[HW_ADDR_TO_STRING_MAX]){0})
```

Comment:
Hmm, I do not like this... Why do not you do like the following? At least, we usually drop 0 in `{}`.
```
char buf[HW_ADDR_TO_STRING_MAX] = {};

log_link_debug("...: %s", hw_addr_to_string(&hw_addr, buf));
```

Reply:
It's purely a convenience macro for cases where you just want to log/printf/... the address, so you can save the explicit buffer declaration.

If the explicit style is preferred, I got no issues with that either.

Reply:
Hmm, I do not have any strong opinion about this. If I remember correctly, our coding style does not prohibit such a implicit declaration, but at least this is the first time I see it in our code.

@poettering and @keszybz WDYT?